tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -699,6 +710,19 @@ public void canLoggedInUserViewOrAmend(Long postId) {
     }
   }
 
+  @Override
+  public void checkTheLoggedInUserDbSameAsPostOwner(Long postId, UserProfile userProfile) {
+    Optional<Post> optionalPost = postRepository.findById(postId);
+    if (optionalPost.isPresent()) {
+      String postOwner = optionalPost.get().getOwner();
+      if (!DesignatedBodyMapper.
+          map(userProfile.getDesignatedBodyCodes()).contains(postOwner)) {
+        throw new AccessUnauthorisedException(
+            "You cannot delete Post with id: " + postId);
+      }
+    }
+  }
```

Comment:
If this is unlikely to be called anywhere other than for deleting, as indicated by the Exception message, we can avoid repeated calls to fetch the Post by including it in the delete operation.

Reply:
Refer to the previous comment in the PostResource interface